smartsim/experiment.py

```diff
@@ -160,28 +168,45 @@ def __init__(
 
         self.exp_path = exp_path
 
-        self._launcher = launcher.lower()
-
-        if self._launcher == "auto":
-            self._launcher = detect_launcher()
-        if self._launcher == "cobalt":
-            raise SSUnsupportedError("Cobalt launcher is no longer supported.")
-
-        if launcher == "dragon":
-            self._set_dragon_server_path()
-
-        self._control = Controller(launcher=self._launcher)
+        # TODO: Remove this! The contoller is becoming obsolete
+        self._control = Controller(launcher="local")
+        self._dispatcher = settings_dispatcher
+        self._active_launchers: set[LauncherLike[t.Any]] = set()
 
         self.fs_identifiers: t.Set[str] = set()
         self._telemetry_cfg = ExperimentTelemetryConfiguration()
 
-    def _set_dragon_server_path(self) -> None:
-        """Set path for dragon server through environment varialbes"""
-        if not "SMARTSIM_DRAGON_SERVER_PATH" in environ:
-            environ["SMARTSIM_DRAGON_SERVER_PATH_EXP"] = osp.join(
-                self.exp_path, CONFIG.dragon_default_subdir
+    def start_jobs(self, *jobs: Job) -> tuple[LaunchedJobID, ...]:
+        """WIP: replacemnt method to launch jobs using the new API"""
+
+        if not jobs:
+            raise TypeError(
+                f"{type(self).__name__}.start_jobs() missing at least 1 required "
+                "positional argument"
             )
 
+        def _start(job: Job) -> LaunchedJobID:
+            builder = job.launch_settings.launch_args
+            launcher_type = self._dispatcher.get_launcher_for(builder)
+            launcher = first(
+                lambda launcher: type(launcher) is launcher_type,
```

Comment:
Exactly! And that way if the experiment already has a shell launcher, it can just reuse the existing one as opposed to standing up a new one for each job!